src/main/kotlin/com/mdrobnak/lalrpop/psi/ext/LpTypeRef.kt

```diff
@@ -0,0 +1,71 @@
+package com.mdrobnak.lalrpop.psi.ext
+
+import com.intellij.extapi.psi.ASTWrapperPsiElement
+import com.intellij.lang.ASTNode
+import com.mdrobnak.lalrpop.psi.*
+import com.mdrobnak.lalrpop.psi.util.isRefMut
+import com.mdrobnak.lalrpop.psi.util.lifetimeOrInfer
+import org.rust.lang.core.psi.ext.childrenWithLeaves
+import org.rust.lang.core.psi.ext.elementType
+
+abstract class LpTypeRefMixin(node: ASTNode) : ASTWrapperPsiElement(node), LpTypeRef {
+    override fun resolveType(arguments: List<NonterminalGenericArgument>): String {
+        return when (childrenWithLeaves.first().elementType) {
+            // tuple
+            LpElementTypes.LPAREN -> this.typeRefList.joinToString(prefix = "(", postfix = ")") {
```

Comment:
Does this need a comma separator?

Reply:
Yes it does but default separator is `", "`. Will add anyway.